### PR TITLE
Bugfix/acpp nvhpc dep type

### PR DESCRIPTION
**Pull request by will-saunders-ukaea**
PR to fix the llvm openmp and llvm cuda backends.
* If there are no backends specified then explicitly set the default backend to omp.library-only
* If there is an llvm backed add the additional rpaths needed to find the libomp.so that the omp.accelerated backend links to
* If there is a cuda backend add the rpath for the llvm+cuda libraries.

### CODE REVIEW

File: packages/adaptivecpp/package.py [hunk 1]
```diff
@@ -218,36 +218,82 @@ def filter_config_file(self):
...
                     "found: {0}".format(so_paths)
                 )
-            rpaths.add(path.dirname(so_paths[0]))
+            rpaths_cpp.add(path.dirname(so_paths[0]))
+
+            # the omp llvm backend may link against the libomp.so in llvm
+            so_paths = filesystem.find(self.spec["llvm"].prefix, "libomp.so")
+            rpaths_cpp.add(path.dirname(so_paths[0]))
+        
+        if "llvm" in self.spec:
```
Comment (cmacmackin) [inline packages/adaptivecpp/package.py:263]: Why is this a separate if-block from the above? They have the same condition.

File: packages/adaptivecpp/package.py [hunk 2]
```diff
@@ -218,36 +218,82 @@ def filter_config_file(self):
...
-            rpaths = set()
             so_paths = filesystem.find(self.spec["llvm"].prefix, "libc++.so")
             if len(so_paths) != 1:
                 raise InstallError(
                     "concretized llvm dependency must provide a "
                     "unique directory containing libc++.so, "
                     "found: {0}".format(so_paths)
                 )
-            rpaths.add(path.dirname(so_paths[0]))
+            rpaths_cpp.add(path.dirname(so_paths[0]))
```
Comment (cmacmackin) [inline packages/adaptivecpp/package.py:249]: Why did you change the name of this variable?

**Review by cmacmackin — COMMENTED** [review]
I think this looks basically good, but there were a couple of things I wasn't clear on. Possibly they were both just the result of the development process.

File: packages/adaptivecpp/package.py (hunk 2 above)
Comment (will-saunders-ukaea) [reply packages/adaptivecpp/package.py]: I was expecting there to be some llvm rpaths and some cuda rpaths and this ended up not being the case. Reverted to just "rpaths".

File: packages/adaptivecpp/package.py (hunk 1 above)
Comment (will-saunders-ukaea) [reply packages/adaptivecpp/package.py]: Thanks, fused these conditionals.

**Review by cmacmackin — APPROVED** [review]
